myVK/myVK/ViewController/Controller/Table/FriendsTableViewController.swift

```diff
@@ -80,17 +83,47 @@ extension FriendsTableViewController {
     }
 }
 
-// MARK: - Network Secrvice Method
+// MARK: - Secrvice Method
 
 extension FriendsTableViewController {
     // MARK: - Private Methods
 
-    private func networkFetchFriends() {
-        networkService.fetchFriends { [weak self] users in
+    private func addNotificationToken(result: Results<User>) {
+        notificationToken = result.observe { [weak self] (changes: RealmCollectionChange) in
             guard let self = self else { return }
-            self.users = users
-            self.alphabetSort()
-            self.tableView.reloadData()
+            switch changes {
+            case .initial:
+                break
+            case .update:
+                self.users = result
+                self.alphabetSort()
+                self.tableView.reloadData()
+            case let .error(error):
+                print(error.localizedDescription)
+            }
+        }
+    }
+
+    private func loadRealmData() {
+        guard let items = RealmService.defaultRealmService.readData(type: User.self) else { return }
+        addNotificationToken(result: items)
+        if !items.isEmpty {
+            users = items
+        } else {
+            networkFetchFriends()
+        }
+        alphabetSort()
+        tableView.reloadData()
+    }
+
+    private func networkFetchFriends() {
+        networkService.fetchFriends { result in
+            switch result {
+            case .success:
+                break
+            case let .failure(error):
+                print(error.localizedDescription)
+            }
```

Comment:
```suggestion
            if case let .failure(error):
                print(error.localizedDescription)
            }
```
и по-хорошему алерт дернуть и отобразить ошибку